### DISCUSSION TITLE
Kernel variables

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#22261] [Q&A Modules: Phase field] [answered]
Dear MOOSE experts,

I am very new to MOOSE and want to model convection flow (for phase field model). I have followed Example 2 and Example 8, but my problem is that variable u in v ⋅∇u is a function of other variables. In this case, how should I define u in ex02.i? Any help would be appreciated. 

Kinds,

Jung

**Comment by permcody** [comment, marked as answer]
Take a look at Example 3: In short, you'll create multiple kernels for your terms in your "u" and "v" equations. If you have a term where both variables appear (i.e. v ⋅∇u), you'll use the "coupling" capability to pull in values from the other variable for use in calculating your term

https://mooseframework.inl.gov/getting_started/examples_and_tutorials/examples/ex03_coupling.html
